.github/workflows/publish-provider.yml

```diff
@@ -0,0 +1,104 @@
+name: Publish Terraform Provider
+
+on:
+  workflow_run:
+    workflows: ["Release"]
+    types:
+      - completed
+
+jobs:
+  publish:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    environment: publish
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Download release assets
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh release download ${{ github.event.workflow_run.head_branch }} --repo ${{ github.repository }} --dir dist/
+
+      - name: Extract short version
+        id: extract_version
+        run: |
+          TAG_NAME="${{ github.event.workflow_run.head_branch }}"
+          SHORT_VERSION=$(echo "$TAG_NAME" | sed -E 's/v([0-9]+\.[0-9]+\.[0-9]+)-.*/\1/')
+          WITHOUT_V=$(echo "$TAG_NAME" | sed -E 's/v//')
+          echo "short_version=$SHORT_VERSION" >> $GITHUB_ENV
+          echo "without_v=$WITHOUT_V" >> $GITHUB_ENV
+
+      - name: Create version.json
+        run: |
+          cat <<EOF > version.json
+          {
+            "data": {
+              "type": "registry-provider-versions",
+              "attributes": {
+                "version": "${{ env.short_version }}",
+                "key-id": "3AAFE0237464B9EB",
```

Comment:
Where does this string come from?

Reply:
After calling TFE API to upload a public key, they return this id in the response. Such IDs don't look like a secret to me, especially because in the docs they show the id of their public Hashicorp public gpg key. This id is not equal to the fingerprint